### DISCUSSION TITLE
Creation of lamellars within daughter grains

### DISCUSSION THREAD

**Discussion by J-AKAZ** [neperfepx/neper#211] [Q&A] [answered]
Hello dears,
I try to create lamellars grains within daughter grains in a microstructure.  This microstructure contains 4 parent grains of vronoi generated from the **coo1** file. Then, the daughter grains are generated from the **refinement.txt** file. However, the generation of this microstructure stops as soon as the lames are generated. The command I used is as follows : 
```
neper -T -n "4::from_morpho::file(raffinement.txt)" -domain 'cube(0.03,0.08,0.03)' -morphooptiini "coo:file(coo1),weight:0" -morpho "voronoi::sphericity:lognormal(0.001,0.0001)::lamellar(w=file(lamwidth.txt))" -oridescriptor 'euler-roe' -format tess,vtk,ori -o D522_3_1
```
Could you help me please ?
Thank you in advance :)

**Comment by rquey** [comment, marked as answer]
Difficult to say without the input files (and your Neper terminal output).

**Reply by J-AKAZ** [reply]
Hello Romain,
In the Neper terminal output i have : 

```
> ========================    N   e   p   e   r    =======================
> Info   : A software package for polycrystal generation and meshing.
> Info   : Version 4.1.2
> Info   : Built with: gsl|muparser|opengjk|openmp|nlopt
> Info   : Running on 12 threads.
> Info   : <https://neper.info>
> Info   : Copyright (C) 2003-2020, and GNU GPL'd, by Romain Quey.
> Info   : No initialization file found (`/home/g20281/.neperrc').
> Info   : ---------------------------------------------------------------
> Info   : MODULE  -T loaded with arguments:
> Info   : [ini file] (none)
> Info   : [com line] -n 4::from_morpho::file(raffinement.txt) -domain
>          cube(0.03,0.08,0.03) -morphooptiini coo::file(coo1.txt),weight:0
>          -morpho
>          vornoi::sphericity:lognormal(0.001,0.0001)::lamellar(w=0.001)
>          -oridescriptor euler-roe -format tess,ori -o D_r
> Info   : ---------------------------------------------------------------
> Info   : Reading input data...
> Error  :   - `-morpho lamellar(w=0.001)' requires `-n from_morpho'.
> Abandon
> 
```
The input files are bellow : 
[rafinnement.txt](https://github.com/rquey/neper/files/6546765/rafinnement.txt)
[coo1.txt](https://github.com/rquey/neper/files/6546796/coo1.txt)
Thanks in advance

**Reply by rquey** [reply]
There are several major issues:
- at scale 2, you are giving `-n from_morpho` but `-morpho` is `sphericity:lognormal(0.001,0.0001)`. The number of grains cannot be determined from a sphericity distribution.
- `sphericity:lognormal(0.001,0.0001)` is insanely small. Is it really what you want?
- at scale 3, you are giving `-n "file(raffinement.txt)"`, but `-morpho` is `"lamellar(w=0.001)"`. This morpho requires `from_morpho`. 
-  the first column of `raffinement.txt` contains standard ids, while they should be multiscale ids of the form `foo::bar` in this case.

You should try to get a proper 2-scale tessellation before going 3-scale.

**Reply by J-AKAZ** [reply]
Thank you for your reply. But i have a question :  Could you explain for me what do you mean (or give me some examples ) by the form foo::bar that should be used in the first column of _raffinement.txt_ ? (I look for that form in the documentation but i didn't find nothing..)
Thank you,

**Reply by rquey** [reply]
_foo_ and _bar_ are [placeholder names](https://en.wikipedia.org/wiki/Foobar). Multiscale ids for scale-2 cells are of the form 1::1, 1::2, 1::3, ..., 2::1, etc.
